### PR TITLE
golint: fix hosting issue

### PR DESCRIPTION
**Pull request by w9n**
Signed-off-by: Robin Winkelewski <w9ncontact@gmail.com>

fixes #3013 CI error

**- What I did**

fix issue of https://golang.org/x/lint being not available

**- How to verify it**

CI

**- Description for the changelog**

golint: fix hosting issue by directly git cloning from github

**- A picture of a cute animal (not mandatory but encouraged)**
monkey

### CODE REVIEW

**Comment by w9n** [comment]
hm https://github.com/golang/lint/issues/397

**Comment by rn** [comment]
I hit the same issue yesterday but it went away. I'm not sure if this is temporary flakiness of golang.org...

In any case a better fix might be to use the alpine base image in CI which has the tools we need already baked in

**Comment by rn** [comment]
there is go in the `linuxkit/alpine`, that's what I meant by "alpine base image".

**Comment by w9n** [comment]
hm i tried this https://circleci.com/gh/linuxkit/linuxkit/800 on linuxkit/alpine:f3cd219615428b2bd943411723eb28875275fae7

but of course you are right its in the packages list: https://github.com/linuxkit/linuxkit/blob/master/tools/alpine/packages

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,7 +16,9 @@ jobs:
       - run:
           name: Dependencies
           command: |
-            go get -u github.com/golang/lint/golint
+            mkdir -p $GOPATH/src/golang.org/x
+            git clone https://github.com/golang/lint.git $GOPATH/src/golang.org/x/lint
```
Comment (justincormack) [inline .circleci/config.yml]: you can do a depth=1 clone here

**Comment by ijc** [comment]
The immediate issue has been fixed at the golang end: https://github.com/golang/lint/issues/397#issuecomment-383997188

Not sure if we still want this -- using the alpine image for the CI base image is an interesting(ish) idea.

**Comment by w9n** [comment]
Giving up with the current linker error, it works locally for me :/

**Pull request closed without merging**